src/libmongoc/tests/TestSuite.c

```diff
@@ -125,8 +162,14 @@ TestSuite_Init (TestSuite *suite, const char *name, int argc, char **argv)
    suite->flags = 0;
    suite->prgname = bson_strdup (argv[0]);
    suite->silent = false;
+   suite->ctest_run = NULL;
    _mongoc_array_init (&suite->match_patterns, sizeof (char *));
 
+   suite->prev_sigabrt = signal (SIGABRT, _handle_signal);
+   suite->prev_sigint = signal (SIGINT, _handle_signal);
+   suite->prev_sigterm = signal (SIGTERM, _handle_signal);
+   suite->prev_sigsegv = signal (SIGSEGV, _handle_signal);
```

Comment:
Is registering signal handlers for `SIGINT`, `SIGTERM`, and `SIGSEGV` necessary? This goes beyond the initial suggestion to use a signal handler to replace existing calls to `abort ()` with `exit (1)` behavior, which only applies to the `SIGABRT` signal.

Reply:
It isn't *necessary*, but it is helpful to see why a particular test is exiting abnormally.